pkg/apis/kubeone/types.go

```diff
@@ -257,6 +258,21 @@ type SystemPackages struct {
 	ConfigureRepositories bool `json:"configureRepositories"`
 }
 
+// PodNodeSelector feature flag
+type PodNodeSelector struct {
+	Enable bool                  `json:"enable"`
+	Config PodNodeSelectorConfig `json:"config"`
+}
+
+// PodNodeSelectorConfig config
+type PodNodeSelectorConfig struct {
+	// ConfigFilePath is a path on the local file system to the PodNodeSelector
+	// configuration file.
+	// ConfigFilePath is a required field.
+	// More info: https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#podnodeselector
+	ConfigFilePath string `json:"configFilePath"`
```

Comment:
```suggestion
	Config string `json:"config"`
```

Let's have it inline instead of file

Reply:
I've thought about this as well and I'm not sure this is a good approach. I think we should have the standardized API, even if it's a little bit redundant sometimes. A couple of things I have on mind:

- `Config` is a struct for all other features. Having it as a string here could introduce confusion
- The problem mentioned above could be solved by calling it `ConfigFilePath` like it's now, but then we come to another problem...
- If we ever have to add another field for any reason, we would drift from the standard API way too much and we would end up in a mess.

In the long run, I think it's better as it is right now.